projects/etos_suite_runner/src/etos_suite_runner/lib/suite.py

```diff
@@ -300,7 +295,13 @@ def sub_suite_environments(self) -> Iterator[dict]:
             ):
                 if environment["meta"]["id"] not in environments:
                     environments.append(environment["meta"]["id"])
-                    yield environment
+                    sub_suite_definition = self._download_sub_suite(environment)
+                    if sub_suite_definition is None:
+                        raise EnvironmentProviderException(
```

Comment:
This method is recording all other exceptions via OpenTelemetry.

Reply:
That would be a change in behavior from before. This part of the code has just been moved from below.